### PR TITLE
201702 mfd for faults part2

### PR DESCRIPTION
**Pull request by mmpagani**
This PR contains the second part of the changes introduced to support shallow fault sources

### CODE REVIEW

File: mbt/importers/faults.py [hunk 1]
```diff
@@ -168,14 +173,21 @@ def get_oq_shp_faults(shapefile_filename, log=False):
...
+            raise ValueError('Sources with non unique ID %s' % sid)
+
     dataSource.Destroy()
 
-    srtd = sorted(sources, key=lambda x: x.source_id, reverse=False)
-    if log:
-        for src in srtd:
-            print '%4s %-40s' % (src.id, src.name)
-    return srtd
+    #srtd = sorted(sources, key=lambda x: x.source_id, reverse=False)
```
Comment (nastasi-oq) [inline mbt/importers/faults.py]: Why not remove it (or at least add a comment to explain the commented block).

File: mbt/importers/faults.py [hunk 2]
```diff
@@ -168,14 +173,21 @@ def get_oq_shp_faults(shapefile_filename, log=False):
 	    value = get_value(mapping[key], feature.GetField(key))
             setattr(src, mapping[key], value)
 
-        sources.append(src)
+        if not id_set and set(sid):
```
Comment (nastasi-oq) [inline mbt/importers/faults.py]: Wrong operator here, I suppose.

File: mbt/notebooks/sources_area/load_data_from_shapefile.ipynb [hunk 3]
```diff
@@ -150,7 +153,836 @@
    "version": "2.7.6"
   },
   "widgets": {
-   "state": {},
+   "state": {
```
Comment (nastasi-oq) [inline mbt/notebooks/sources_area/load_data_from_shapefile.ipynb]: Uncleaned state.

File: mbt/notebooks/sources_shallow_fault/load_data_from_shapefile.ipynb [hunk 4]
```diff
@@ -0,0 +1,673 @@
...
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython2",
+   "version": "2.7.6"
+  },
+  "widgets": {
+   "state": {
```
Comment (nastasi-oq) [inline mbt/notebooks/sources_shallow_fault/load_data_from_shapefile.ipynb]: Uncleaned state.

File: mbt/tools/faults.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+    # Compute total rate
+    rate_above = _get_rate_above_m_low(moment_from_slip, m_low, m_upp)
+
+    # Compute rate per bin
+    rrr = []
+    mma = []
+    for mmm in numpy.arange(m_low, m_upp, bin_width):
+        rte = (_get_cumul_rate_truncated(mmm, m_low, m_upp, rate_above, b_gr) -
+               _get_cumul_rate_truncated(mmm+bin_width, m_low,
```
Comment (nastasi-oq) [inline mbt/tools/faults.py]: Spaces around operators.

File: mbt/notebooks/sources_area/load_data_from_shapefile.ipynb (hunk 3 above)
Comment (mmpagani) [inline mbt/notebooks/sources_area/load_data_from_shapefile.ipynb]: Yes you're right. The new version should now be clean.

**Comment by mmpagani** [comment]
Hopefully the issues should be all be fixed

**Comment by mmpagani** [comment]
Code seems clean to me

**Pull request merged**